src/baUtility.cpp

```diff
@@ -0,0 +1,16 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+NumericVector baUtility_rcpp(double aBA, double bBA, NumericVector BA, IntegerVector idx_BA) {
+  NumericVector utility (33);
+    
+  int k = BA.length();
+  
+  for(int i = 0; i < k; i++) {
+    int idx = idx_BA[i];
+    utility[idx] = bBA / pow(std::max(BA[i], 0.0), aBA);

```

Comment:
Coming back to 3, it's possible that actually std::pow already has a case built-in for 0, so then leaving it as is would be better. Still, the general principle of trying to avoid pointless computations is good to keep in mind when optimizing :)